### DISCUSSION TITLE
How to check if an image is already exist in cache (file or memcache)

### DISCUSSION THREAD

**Discussion by yatpannggithub** [coil-kt/coil#1919] [General]
When loading images, the designer wanted to use a placeholder BitmapDrawable that is computationally expensive to render (e.g. Bitmap from Blurhash). 

I want to avoid creating the bitmap if the full image is already cached (i.e. if placeholder will not be used, there is no point in creating the bitmap) How can I check if a key is cached in memory or not since MemoryCache.Key cannot be created manually.

```
        val cacheKey = _uri.toString()
        val isInDiskCache = try { context.imageLoader.diskCache?.openSnapshot(cacheKey)?.use { true } ?: false } catch (t: Throwable) { false }
 
        val isInMemCache = context.imageLoader.memoryCache?.let {
            it.get(MemoryCache.Key(cacheKey)) != null
        }

```

**Comment by yatpannggithub** [comment]
I end up just checking disk cache and ignore mem cache for now. So end up using this
```
    return try { context.imageLoader.diskCache?.get(cacheKey)?.use {
        it.data.toFile().exists()
    } ?: false } catch (t : Throwable) { false }
```

**Comment by yschimke** [comment]
FWIW I just came to request something similar.  I found I was incorrectly using get() without closing the resource.  But I don't actually need to open it.  I just want to know if it exists, before prefetching images.
